src/main/java/org/opensearch/knn/index/codec/KNN990Codec/NativeEngines990KnnVectorsWriter.java

```diff
@@ -244,26 +244,25 @@ private <T, C> void trainAndIndex(
         final String operationName
     ) throws IOException {
         final VectorDataType vectorDataType = extractVectorDataType(fieldInfo);
-        KNNVectorValues<T> knnVectorValues = vectorValuesRetriever.apply(vectorDataType, fieldInfo, VectorProcessingContext);
-        QuantizationParams quantizationParams = quantizationService.getQuantizationParams(fieldInfo);
-        QuantizationState quantizationState = null;
         // Count the docIds
         int totalLiveDocs = getLiveDocs(vectorValuesRetriever.apply(vectorDataType, fieldInfo, VectorProcessingContext));
-        if (quantizationParams != null && totalLiveDocs > 0) {
+        if (totalLiveDocs == 0) {
+            log.debug("No live docs for field " + fieldInfo.name);
```

Comment:
Just to confirm - this wont lead to an issue on search if an empty file is expected?

Reply:
@jmazanec15 Can you give more context when do we expect empty file? I see this is already added inside index method https://github.com/opensearch-project/k-NN/blob/7bf52ac59f469e7e4a713513070a54d59e00e966/src/main/java/org/opensearch/knn/index/codec/nativeindex/NativeIndexWriter.java#L134

Reply:
For search and warmup, we should make sure that we arent reading this file if the size is empy

Reply:
Discussed offline and confirmed that if file is empty, we are not reading the file since we check whether file exists or not before reading it.